### DISCUSSION TITLE
Difference between `on_batch_start` and `on_train_batch_start`?

### DISCUSSION THREAD

**Discussion by daniellepintz** [Lightning-AI/pytorch-lightning#9727] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule]
What is the difference between `on_batch_start` and `on_train_batch_start`? Same question for `on_batch_end` and `on_train_batch_end`.

https://github.com/PyTorchLightning/pytorch-lightning/blob/49d03f87fed0458cbb146d38243be56be4cb9689/pytorch_lightning/trainer/callback_hook.py#L155-L173

If they are different, can we make the docstrings different?

If they are the same, can we deprecate one of them?

**Comment by rohitgr7** [comment]
I remember this was discussed a few months back, on_batch_* was supposed to run for all train/valid/test/predict, but looks like it just runs in case of training. Looks redundant now in case there are no plans to call it irrespective of running_stage.

**Reply by daniellepintz** [reply]
thanks @rohitgr7! that's what I thought too. Does anyone have any preference of either deprecation or making it work as intended (run for all train/valid/test/predict)? @ananthsub @tchaton @carmocca @awaelchli @SeanNaren @kaushikb11

**Reply by carmocca** [reply]
It should work as intended. There are several TODOs [in this file](https://github.com/PyTorchLightning/pytorch-lightning/blob/fa44dbcd9e4c6d5fd628ad2f870bbff620a3f0b2/tests/models/test_hooks.py#L290) pointing the inconsistencies.

**Reply by daniellepintz** [reply]
I see, thanks! Is there something blocking us from doing those TODOs?  Also should we update the docstrings to """Called when a batch begins (train/valid/predict/test).""

**Reply by carmocca** [reply]
> Is there something blocking us from doing those TODOs

Nothing! Feel free to create an issue for us to discuss/approve and then work on fixing them.

> should we update the docstrings

Yes
